numpy/distutils/fcompiler/environment.py

```diff
@@ -55,9 +55,11 @@ def _get_var(self, name, conf_desc):
         if envvar is not None:
             envvar_contents = os.environ.get(envvar)
             if envvar_contents is not None:
-                if var and append:
+                if convert:
+                    envvar_contents = convert(envvar_contents)
```

Comment:
Simpler to put at the top of the function:
```
if convert is None:
    convert = lambda x: x
```

Then you can remove the `if` here and below.